### PR TITLE
Reduce use of statics. Prep for non-static SerializationManager

### PR DESCRIPTION
**Pull request by ReubenBond**
Simple changes which were extracted from #2592.

The changes are very basic. Some of the changes only make sense if you evaluate them with the understanding that they are in preparation for making SerializationManager non-static, so keep that in mind.

Most of the changes are simply passing fixtures into test classes.

### CODE REVIEW

**Comment by ReubenBond** [comment]
@dotnet-bot test this please

File: src/OrleansCodeGenerator/SerializerGenerationManager.cs [hunk 1]
```diff
@@ -117,7 +117,7 @@ internal bool RecordTypeToGenerate(Type t, Module module, Assembly targetAssembl
             return true;
         }
 
-        private static bool IsFieldInaccessibleForSerialization(Module module, Assembly targetAssembly, FieldInfo field)
```
Comment (benjaminpetit) [inline src/OrleansCodeGenerator/SerializerGenerationManager.cs]: >static [](start = 16, length = 6)

Really needed? Or just in preparation of a next PR?

Comment (ReubenBond) [inline src/OrleansCodeGenerator/SerializerGenerationManager.cs]: It's prep, see the call to `SerializationManager.HasSerialzier`? We know a field can be accessed for serialization if we already have a serializer for it

File: src/OrleansAzureUtils/Storage/Exceptions.cs [hunk 2]
```diff
@@ -48,7 +48,7 @@ public class TableStorageUpdateConditionNotSatisfiedException : InconsistentStat
         /// <summary>
         /// Id of grain
         /// </summary>
-        public GrainReference GrainId { get; }
+        public string GrainId { get; }
```
Comment (sergeybykov) [inline src/OrleansAzureUtils/Storage/Exceptions.cs]: >public string GrainId { get; } [](start = 8, length = 30)

Why do we need to switch from `GrainReference` to `string` here? Seems unrelated to elimination of statics.

**Comment by sergeybykov** [comment]
Except for the question above LGTM.

File: test/NonSiloTests/General/Identifiertests.cs [hunk 3]
```diff
@@ -12,19 +12,20 @@
...
     {
         private readonly ITestOutputHelper output;
+        private readonly TestEnvironmentFixture environment;
         private static readonly Random random = new Random();
 
         class A { }
         class B : A { }
         
-        public Identifiertests(ITestOutputHelper output)
+        public Identifiertests(ITestOutputHelper output, TestEnvironmentFixture fixture)
```
Comment (jdom) [inline test/NonSiloTests/General/Identifiertests.cs]: same, no need to add the fixture to the constructor if you do not intend to access it from the tests

File: src/OrleansAzureUtils/Storage/Exceptions.cs (hunk 2 above)
Comment (ReubenBond) [inline src/OrleansAzureUtils/Storage/Exceptions.cs]: It's because GrainReferences internally need a reference to `IRuntimeClient`, which BinaryFormatterSerializer can't provide. We need that info when calling `FromKeyString` below.

File: src/OrleansRuntime/Messaging/IncomingMessageAcceptor.cs [hunk 4]
```diff
@@ -401,7 +402,7 @@ private SocketAsyncEventArgs GetSocketReceiveAsyncEventArgs(Socket sock)
             return saea.SocketAsyncEventArgs;
         }
 
-        private static SaeaPoolWrapper CreateSocketReceiveAsyncEventArgsPoolWrapper()
+        private SaeaPoolWrapper CreateSocketReceiveAsyncEventArgsPoolWrapper()
```
Comment (ReubenBond) [inline src/OrleansRuntime/Messaging/IncomingMessageAcceptor.cs]: It is, but not in this PR

File: test/NonSiloTests/General/InternalSerializationTests.cs [hunk 5]
```diff
@@ -15,14 +15,17 @@ namespace UnitTests.Serialization
...
+    [Collection(TestEnvironmentFixture.DefaultCollection)]
     public class InternalSerializationTests
     {
+        private readonly TestEnvironmentFixture fixture;
+
         /// <summary>
         /// Initializes the system for testing.
         /// </summary>
-        public InternalSerializationTests()
+        public InternalSerializationTests(TestEnvironmentFixture fixture)
```
Comment (ReubenBond) [inline test/NonSiloTests/General/InternalSerializationTests.cs]: These will all be used, since SerializationManager will be on that fixture

File: test/DefaultCluster.Tests/GrainReferenceTest.cs [hunk 6]
```diff
@@ -117,33 +117,32 @@ private static void TestGrainReferenceSerialization(int id, bool resolveBeforeSe
...
-            Assert.Equal(grain,  otherGrain);  // "Deserialized grain reference equality is preserved"
-            int res = otherGrain.GetA().Result;
+            Assert.NotNull(other);
+            Assert.Equal(grain,  other);  // "Deserialized grain reference equality is preserved"
+            int res = other.GetA().Result;
             Assert.Equal(id,  res);  // "Returned values from call to deserialized grain reference"
         }
 
-        private static object DotNetSerialiseRoundtrip(object obj)
+        private static T DotNetSerialiseRoundtrip<T>(T obj)
```
Comment (ReubenBond) [inline test/DefaultCluster.Tests/GrainReferenceTest.cs]: fixed
